core/src/main/java/com/sequenceiq/distrox/v1/distrox/fedramp/FedRAMPModificationService.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.distrox.v1.distrox.fedramp;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.common.gov.CommonGovService;
+import com.sequenceiq.cloudbreak.common.provider.ProviderPreferencesService;
+import com.sequenceiq.distrox.api.v1.distrox.model.DistroXV1Request;
+import com.sequenceiq.distrox.api.v1.distrox.model.database.DistroXDatabaseAvailabilityType;
+import com.sequenceiq.distrox.api.v1.distrox.model.database.DistroXDatabaseRequest;
+
+@Service
+public class FedRAMPModificationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FedRAMPModificationService.class);
+
+    @Inject
+    private CommonGovService commonGovService;
+
+    @Inject
+    private ProviderPreferencesService preferencesService;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    public void prepare(DistroXV1Request request, String accountId) {
+        boolean govCloudDeployment = commonGovService.govCloudDeployment(
+                preferencesService.enabledGovPlatforms(),
+                preferencesService.enabledPlatforms());
+        if (govCloudDeployment) {
+            LOGGER.info("The current deployment is a fedramp deployment so needs to modify the distrox request to use external database");
+            if (!entitlementService.isFedrampExternalDatabaseForceDisabled(accountId)) {
+                LOGGER.info("CDP_FEDRAMP_EXTERNAL_DATABASE_FORCE_DISABLED NOT applied on the tenant so adding HA external database to the distrox request");
+                DistroXDatabaseRequest distroXDatabaseRequest = new DistroXDatabaseRequest();
+                distroXDatabaseRequest.setAvailabilityType(DistroXDatabaseAvailabilityType.HA);
+                request.setExternalDatabase(distroXDatabaseRequest);
```

Comment:
First I would check if the DistroX request has a DistroXDatabase request and create a new one only if missing.
In the case when it exists I would just update the AvailabiltyType